### PR TITLE
Code style consistency improved.

### PR DESCRIPTION
**Pull request by chernecov**
Code style consistency improved a little bit.
No major changes, few exception were added, reason - consistency between code and returning types.

### CODE REVIEW

File: Representation/PaginatedRepresentation.php [hunk 1]
```diff
@@ -109,12 +110,15 @@ public function hasNextPage()
...
      * @return int
+     * @throws \Exception
      */
     public function getNextPage()
     {
         if ($this->hasNextPage()) {
             return $this->page + 1;
         }
+
+        throw new \Exception('Next page does not exist');
```
Comment (RuslanZavacky) [inline Representation/PaginatedRepresentation.php]: I am not sure that we should throw an exception, if there is no new page, the result should be empty. If we throw an exception, then we should handle response always with try/catch

File: Representation/PaginatedRepresentation.php [hunk 2]
```diff
@@ -127,11 +131,14 @@ public function hasPreviousPage()
...
      * @return int
+     * @throws \Exception
      */
     public function getPreviousPage()
     {
         if ($this->hasPreviousPage()) {
             return $this->page - 1;
         }
+
+        throw new \Exception('Previous page does not exist');
```
Comment (RuslanZavacky) [inline Representation/PaginatedRepresentation.php]: Same here :)

File: Representation/PaginatedRepresentation.php (hunk 1 above)
Comment (chernecov) [reply Representation/PaginatedRepresentation.php]: I've added it due to returning type.
If you hint that integer will be returned, then exception should be thrown if nothing to return.
Retunring zero/one as default - sounds bad to me.
We could actually hint int|null - but I don't like it too :)

@RuslanZavacky What will you recommend to return here?

Comment (RuslanZavacky) [reply Representation/PaginatedRepresentation.php]: I'd suggest checking the context where it is used:

```php
'next' => $representation->hasNextPage() ? $this->getUriForPage($representation->getNextPage()) : null,
```
It is fine here to just return number or null. As  `hasNextPage` page check, checks for:

```php
return $this->page < $this->getPages();
```

File: Serializer/JsonApiSerializationVisitor.php [hunk 3]
```diff
@@ -37,19 +34,22 @@ class JsonApiSerializationVisitor extends JsonSerializationVisitor
...
      * @param MetadataFactoryInterface        $metadataFactory
-     * @param                                 $showVersionInfo
+     * @param bool                            $showVersionInfo
      */
     public function __construct(
         PropertyNamingStrategyInterface $propertyNamingStrategy,
         AccessorStrategyInterface $accessorStrategy = null,
         MetadataFactoryInterface $metadataFactory,
-        $showVersionInfo
+        $showVersionInfo = true
```
Comment (RuslanZavacky) [inline Serializer/JsonApiSerializationVisitor.php:50]: I think it is better to be as `false`, actually. It has no big use in response :)

File: Representation/PaginatedRepresentation.php (hunk 1 above)
Comment (chernecov) [reply Representation/PaginatedRepresentation.php]: Deal. Done.

File: Representation/PaginatedRepresentation.php (hunk 2 above)
Comment (chernecov) [reply Representation/PaginatedRepresentation.php]: Deal. Done.

File: Serializer/JsonApiSerializationVisitor.php (hunk 3 above)
Comment (chernecov) [reply Serializer/JsonApiSerializationVisitor.php:50]: @RuslanZavacky I pointed in in your pull request to change it to true, due to bundle configuration.
From configuration as default "true" is passed.

**Comment by koemeet** [comment]
Looks good to me.

**Comment by koemeet** [comment]
Thanks 💯

**Pull request merged**